src/client/auth/oidc.rs

```diff
@@ -197,10 +197,11 @@ pub(crate) async fn authenticate_stream(
     }
 }
 
-async fn update_oidc_cache(
+async fn update_caches(
+    conn: &Connection,
     credential: &Credential,
     response: &IdpServerResponse,
-    token_gen_id: i32,
```

Comment:
spec requires caching the IdP info to avoid asking the server for it when it is known (cached).